src/main.rs

```diff
@@ -218,17 +217,21 @@ struct Args {
 async fn main() -> anyhow::Result<()> {
     let args: Args = Args::parse();
 
-    tracing_subscriber::registry()
-        .with(tracing_subscriber::fmt::layer().pretty().without_time())
-        .with(
-            tracing_subscriber::EnvFilter::try_from_default_env().unwrap_or_else(|_| {
-                tracing_subscriber::EnvFilter::new(format!(
-                    "{},cranelift_codegen=warn",
-                    args.log_level
-                ))
-            }),
-        )
-        .init();
+    let env_filter = tracing_subscriber::EnvFilter::try_from_default_env().unwrap_or_else(|_| {
+        tracing_subscriber::EnvFilter::new(format!("{},cranelift_codegen=warn", args.log_level))
+    });
+
+    if args.enable_structured_logging {
+        tracing_subscriber::registry()
+            .with(tracing_subscriber::fmt::layer().json().without_time())
```

Comment:
I think there are some other values that need to be set, or we run into weird terminal issues as well as parsing issues from the tracing crate at runtime. See https://github.com/wasmCloud/wasmCloud/blob/main/crates/provider-sdk/src/provider_main.rs#L400

Reply:
Awesome thank you, I knew I was missing something

Reply:
Discussed offline: we haven't reproduced issues with interleaving lines (yet?), so we think this is good to go. If we run into issues we can use an approach like above to resolve it

Reply:
I'm going to push this off for the future because I don't think I'm introducing anything different in this PR other than just the basic inclusion of JSON logging, but if we see interleaved logs again we can file an issue + fix